engine/execution/provider/engine.go

```diff
@@ -202,39 +230,36 @@ func (e *Engine) onChunkDataRequest(
 	if sinceProcess > e.chdpQueryTimeout {
 		lg.Warn().Msgf("chunk data pack query takes longer than %v secs", e.chdpQueryTimeout.Seconds())
 	}
-
-	lg.Debug().Msg("chunk data pack response lunched to dispatch")
+	lg.Info().Msg("chunk data pack response lunched to dispatch")
 
 	// sends requested chunk data pack to the requester
-	e.unit.Launch(func() {
-		deliveryStart := time.Now()
+	deliveryStart := time.Now()
 
-		err := e.chunksConduit.Unicast(response, originID)
+	err = e.chunksConduit.Unicast(response, request.RequesterId)
```

Comment:
https://github.com/onflow/flow-go/pull/2951/commits/63617c466dfbc5898cbbe6c12a961e2981eb8770

Reply:
I feel the metrics for querying and delivering chdp is a bit too granular to be useful.

Just like in a web app, we probably care more about the total response time of a request, rather than the individule durations of saving the data to database and sending the message over network. If the total response time is not higher than a threshold, we might not look into those more granular metrics. In other words, we might only look into those granular metrics if the total response time is too high, but in that case, i think we are probably debugging problems, and logs would be more efficient in debugging, as we can associate a certain high response time to a chunk data pack id, and see whether it's due to saving to db takes to long, or delivering over network takes too long.

That being said, IMO, a metrics for total response time of a chunk data pack is enough, and we could log the db query and network delivery along with chdp ID for debugging purpose.

Reply:
👍  Good point Leo. I agree, one metric for total response time seems sufficient. The logs contain more granular timing, and more info about the request, so they're better suited for investigating outliers, which is what the original "log when exceeding the _timeout_" behaviour was geared toward.

Reply:
https://github.com/onflow/flow-go/pull/2951/commits/0f10739e8ecdcfd85c424158a279f803239b2885